### PR TITLE
Fix expectation to exclude self-reference

### LINKED ISSUE #1261: Headers attribute specified on a cell refers to cells in the same table element [a25f45]: Problem with failed example 2
[Failed example 2](https://act-rules.github.io/rules/a25f45#failed-example-2) has a cell with a headers attribute referring to itself, and this is the reason why it fails.
```html
<td id="headerBday" headers="headerBday">
```
However, nothing in the expectation says that the ID have to refer to a **different** cell:
> each of which is the ID of an element, that is a cell of the same table as the target element.

I agree that referring to itself is rather bad (notably because self-referencing are removed by the assignation algorithm).
Thus, the rule should probably be updated to specify that the ID must be of a cell different from the one the test target is set on.

### PR DESCRIPTION
**Pull request by Jym77**
Exclude the target itself from the valid `headers` IDs.
This follow assignation algorithm removing the principal cell from the list of headers, and makes Failed Example 2 actually fail.

Closes issue(s):

- closes #1261

Need for Final Call:
This will require a 1 week Final Call (small update to expectation)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: _rules/table-headers-attribute-refer-to-data-cells-a25f45.md [hunk 1]
```diff
@@ -30,7 +30,9 @@ This rule applies to any `headers` attribute specified on a [`cell`][] within a
 
 ## Expectation
 
-Each target attribute is [a set of space separated IDs][], each of which is the ID of an element, that is a [`cell`][] of the same [`table`][] as the target element.
+Each target attribute is [a set of space separated IDs][], each of which is the ID of an element, that is a [`cell`][] of the same [`table`][] as the target element different from the target.
```
Comment (WilcoFiers) [inline _rules/table-headers-attribute-refer-to-data-cells-a25f45.md:33]: I think this should be handled in a separate expectation.

Comment (Jym77) [reply _rules/table-headers-attribute-refer-to-data-cells-a25f45.md:33]: Yes, that will be easier to read 👍

**Comment by kasperisager** [comment]
By the way, are we sure we want something like `headers="other self"` to fail as well?

**Comment by Jym77** [comment]
> By the way, are we sure we want something like `headers="other self"` to fail as well?

I assumed so because the rule was failing `headers="other INVALID"`. So, I kept the "a single incorrect headers is bad". I agree that it is a bit different, but for example the header assignation is going to treat them the same way (i.e. ignore them).

**Comment by kasperisager** [comment]
Valid point 👍

**Comment by carlosapaduarte** [comment]
I'm not really convinced that `headers="other self"` should fail. If the `self` is removed by the assigning algorithm we are left with `headers="other"`, which passes.
The `headers="other Invalid"` example is different because you will have an invalid reference. 
I would agree to an expectation that checked the list is not composed by self references only, but if there are other valid references in the list I don't think this fails the SC.

**Comment by Jym77** [comment]
> I'm not really convinced that `headers="other self"` should fail. If the `self` is removed by the assigning algorithm we are left with `headers="other"`, which passes.
> The `headers="other Invalid"` example is different because you will have an invalid reference.
> I would agree to an expectation that checked the list is not composed by self references only, but if there are other valid references in the list I don't think this fails the SC.

For the [headers assignation algorithm](https://html.spec.whatwg.org/multipage/tables.html#algorithm-for-assigning-header-cells), "self" and "invalid" are treated exactly the same way (Step 3, first case, point 2):
> For each token in the id list, if the first element in the Document with an ID equal to the token is a cell in the same table, and that cell is not the principal cell, then add that cell to header list.

Given that they are both explicitly ignored by the specs, I do not see why the rule should treat them differently.

I do agree that maybe the expectation should be "at least one is not invalid/self" rather than "none is invalid/self".

**Comment by Jym77** [comment]
Final Call ends on April 30.

**Comment by Jym77** [comment]
Final Call has ended. Merging.

**Pull request merged**